Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  TestTransactionEvent,
+  MockEthersProvider,
+} from "forta-agent-tools/lib/test";
+import { createAddress } from "forta-agent-tools";
+import { provideHandleGovernanceTransaction } from "./agent";
+import {
+  ethers,
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+} from "forta-agent";
+import { ASSET_INFO, CONFIGURATOR_PROXY } from "./constants";
+import * as networkManager from "./networkManager";
+
+describe("Compound test suite", () => {
+  let handleTransaction: HandleTransaction;
+  let mockProvider: MockEthersProvider;
+  let txEvent: TestTransactionEvent;
+  const mockAssetTokenAddress = createAddress("0x10957");
+  const mockArgs = [
+    createAddress("0x1421"),
+    createAddress("0x123"),
+    createAddress("0x234"),
+  ];
+  const Iface = new ethers.utils.Interface(ASSET_INFO);
+  const getCollateralName = new ethers.utils.Interface([
+    "function name() view returns (string)",
+  ]);
+
+  const setupMockProvider = (collateralAddresses: string[]) => {
+    mockProvider.setNetwork(1);
+    collateralAddresses.forEach((address, index) => {
+      mockProvider.addCallTo(mockAssetTokenAddress, 0, Iface, "getAssetInfo", {
+        inputs: [index],
+        outputs: [
+          index,
+          address,
+          createAddress("0x234"),
+          4,
+          12,
+          134,
+          1245,
+          1265,
+        ],
+      });
+      mockProvider.addCallTo(address, 0, getCollateralName, "name", {
+        inputs: [],
+        outputs: [`Collateral-${index + 1}`],
+      });
+    });
+  };
+
+  beforeEach(() => {
+    mockProvider = new MockEthersProvider() as any;
+    const provider = mockProvider as unknown as ethers.providers.Provider;
+    handleTransaction = provideHandleGovernanceTransaction(
+      provider,
+      ASSET_INFO,
+    );
+    txEvent = new TestTransactionEvent().setBlock(0);
+    jest
+      .spyOn(networkManager, "getAddress")
+      .mockResolvedValue(mockAssetTokenAddress);
+    jest
+      .spyOn(networkManager, "getConfigurator")
+      .mockResolvedValue(CONFIGURATOR_PROXY);
```

Comment:
If you update to set up `NetworkManager` to be set up in the `initialize()`, then this could be updated.